PP_APP/server/pp_api/api/views.py

```diff
@@ -25,39 +32,96 @@
 # @permission_classes((IsAuthenticated,))
 @permission_classes(())
 def upload_imgs(request):
-    # upload_model = image_post(user_id=request.user)
-    upload_model = image_post(user_id=Account.objects.get(pk=2))
-    serializer = UploadSerializer(upload_model, data=request.data)
-    data = {}
-    if serializer.is_valid():
-        ret_img_inst = serializer.save()
-        post_id = ret_img_inst.id
+    # upload_model = ImagePost(user_id=request.user)
+
+    """ Check if the email address is valid """
+    normalized_email=BaseUserManager.normalize_email(request.data.get('email'))
+    try:
+        validate_email(normalized_email)
+    except:
+        return Response({'response':'Invalid email address.'}, status=status.HTTP_400_BAD_REQUEST)
+
+    """ 1. Check if user exists. If not create a new user.
+        2. Retreive the user_id
+    """
+    try:
+        user_id = PPUsers.objects.get(email=normalized_email)
+    except PPUsers.DoesNotExist:
+        email_serializer = PPUsersSerializer(data={'email':normalized_email})
+        if email_serializer.is_valid():
+            user_id=email_serializer.save()
+        else:
+            return Response({'response':'Error.'}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
So are people not going to be able to submit images without having a valid email? Could we make accounts optional for now?

Reply:
Any post according to our model should have an email address associated with it. So we cannot make this optional anymore.